Firmware/_daq_core/fir_decimate.c

```diff
@@ -131,10 +137,16 @@ int main(int argc, char **argv)
     
      /* Initializing input shared memory interface */
     struct shmem_transfer_struct* input_sm_buff = calloc(1, sizeof(struct shmem_transfer_struct));
-    if((config.cpi_size*dec)>=config.cal_size)
-    {input_sm_buff->shared_memory_size = config.cpi_size*config.num_ch*dec*4*2+IQ_HEADER_LENGTH;}
+
+    if((config.cpi_size * dec) >= config.cal_size)
+    {
+        input_sm_buff->shared_memory_size = config.cpi_size * dec * config.num_ch * sizeof(uint32_t) * 2 + IQ_HEADER_LENGTH;

```

Comment:
Just a note

As you can here we have fixed bug, because on MacOS if shared memory size do not equal it did not opened